app/src/main/java/com/obsidian/plugins/task_notifier/os/NotificationManager.kt

```diff
@@ -61,22 +57,31 @@ class NotificationManager {
 
     private fun createNotification(
       context: Context,
-      intent: Intent,
       title: String,
       text: String,
-      channel: String
+      scope: ScopeEnum
     ): Notification {
+      val channel: String = this.getChannelFromScope(scope)
+      this.ensureNotificationChannelExists(context, channel)
+      val intent = getPendingIntent(context, scope)
+
       val pendingIntent =
         PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_IMMUTABLE)
+
       val builder: NotificationCompat.Builder =
         NotificationCompat.Builder(context, channel)
-          .setSmallIcon(R.mipmap.ic_launcher)
+          .setSmallIcon(R.mipmap.notification_icon_foreground)
           .setContentIntent(pendingIntent)
-          .setBadgeIconType(R.mipmap.ic_launcher)
+          .setBadgeIconType(BADGE_ICON_LARGE)
+          .setColor(R.color.red)
           .setContentTitle(title)
           .setContentText(text)
           .setPriority(NotificationCompat.PRIORITY_DEFAULT)
 
+      when (scope) {
```

Comment:
if statement is better if we want to do specific things for only one type of data instead of switch case